source/val/validate_atomics.cpp

```diff
@@ -225,6 +225,16 @@ spv_result_t AtomicsPass(ValidationState_t& _, const Instruction* inst) {
                       "be: Uniform, Workgroup, Image, StorageBuffer, or "
                       "PhysicalStorageBuffer.";
           }
+
+          // Can't use result_type because OpAtomicStore doesn't have a result
```

Comment:
This pass leaves a lot to be desired, but this check seems like it would be better located by splitting the else if at line 99 into separate checks for each opcode and putting OpAtomicStore there.

Reply:
well it would involve moving the 

```cpp
      uint32_t operand_index =
          opcode == SpvOpAtomicFlagClear || opcode == SpvOpAtomicStore ? 0 : 2;
      const uint32_t pointer_type = _.GetOperandTypeId(inst, operand_index++);
      if (!_.GetPointerTypeInfo(pointer_type, &data_type, &storage_class)) {
```

logic above that to get the `data_type`

...I do agree, this pass is currently a little rough in some spots

Reply:
It might be easiest to introduce a lambda function that gets the data type and/or bitwidth and then centralize some of these checks instead.

Reply:
my worry is this MR is going to turn into a larger refactor and blow up the scope of the original intent. I would offer to just attempt some amount of refactoring post-this-commit as after trying to add this, up to you, happy to do either route but asking you as you are the one who will end up reviewing it

Reply:
If you want to refactor after this, I'm ok. I'll sign off on this and wait for more clean up in another PR.